pkg/cloudprovider/providers/gce/gce_op.go

```diff
@@ -89,20 +92,89 @@ func getErrorFromOp(op *compute.Operation) error {
+			op, err := gce.serviceAlpha.RegionOperations.Get(gce.projectID, region, operationName).Do()
+			return alphaOperationToV1(op), err
+		}, mc)
+	case *compute_beta.Operation:
+		return gce.waitForOp(betaOperationToV1(op.(*compute_beta.Operation)), func(operationName string) (*compute_v1.Operation, error) {
+			op, err := gce.serviceBeta.RegionOperations.Get(gce.projectID, region, operationName).Do()
+			return betaOperationToV1(op), err
+		}, mc)
+	case *compute_v1.Operation:
+		return gce.waitForOp(op.(*compute_v1.Operation), func(operationName string) (*compute_v1.Operation, error) {
+			return gce.service.RegionOperations.Get(gce.projectID, region, operationName).Do()
+		}, mc)
+	default:
+		return fmt.Errorf("Unexpected Type: %T", v)
+	}
+}
+
+func (gce *GCECloud) waitForZoneOp(op APIObject, zone string, mc *metricContext) error {
+	switch v := op.(type) {
+	case *compute_alpha.Operation:
+		return gce.waitForOp(alphaOperationToV1(op.(*compute_alpha.Operation)), func(operationName string) (*compute_v1.Operation, error) {
+			op, err := gce.serviceAlpha.ZoneOperations.Get(gce.projectID, zone, operationName).Do()
+			return alphaOperationToV1(op), err
+		}, mc)
+	case *compute_beta.Operation:
+		return gce.waitForOp(betaOperationToV1(op.(*compute_beta.Operation)), func(operationName string) (*compute_v1.Operation, error) {
+			op, err := gce.serviceBeta.ZoneOperations.Get(gce.projectID, zone, operationName).Do()
+			return betaOperationToV1(op), err
+		}, mc)
+	case *compute_v1.Operation:
+		return gce.waitForOp(op.(*compute_v1.Operation), func(operationName string) (*compute_v1.Operation, error) {
+			return gce.service.ZoneOperations.Get(gce.projectID, zone, operationName).Do()
+		}, mc)
+	default:
+		return fmt.Errorf("Unexpected Type: %T", v)
+	}
+}
+
+func betaOperationToV1(op *compute_beta.Operation) *compute_v1.Operation {
+	enc, err := op.MarshalJSON()
+	if err != nil {
+		panic(fmt.Sprintf("Failed to encode to json: %v", err))
+	}
+	return convertToV1Operation(enc)
 }
 
-func (gce *GCECloud) waitForRegionOp(op *compute.Operation, region string, mc *metricContext) error {
-	return gce.waitForOp(op, func(operationName string) (*compute.Operation, error) {
-		return gce.service.RegionOperations.Get(gce.projectID, region, operationName).Do()
-	}, mc)
+func alphaOperationToV1(op *compute_alpha.Operation) *compute_v1.Operation {
+	enc, err := op.MarshalJSON()
+	if err != nil {
+		panic(fmt.Sprintf("Failed to encode to json: %v", err))
+	}
+	return convertToV1Operation(enc)
 }
 
-func (gce *GCECloud) waitForZoneOp(op *compute.Operation, zone string, mc *metricContext) error {
-	return gce.waitForOp(op, func(operationName string) (*compute.Operation, error) {
-		return gce.service.ZoneOperations.Get(gce.projectID, zone, operationName).Do()
-	}, mc)
+func convertToV1Operation(enc []byte) *compute_v1.Operation {
```

Comment:
Curious when should we convert objects to V1? Upon the conversion we lost alpha/beta specific fields , so we might not convert objects when doing the real alpha/beta call?
```
type betaObject struct {
    enableBetaFeature bool
    v1Field string
}

type v1Object struct {
    v1Field string
}

func createObject() {
    obj := &betaObject{enableBetaFeature: true}
    // Beta field is dropped so it has no effect?
    createdObject, err := client.create(convertToV1Object(obj))
    ...
}
```